openvdb/openvdb/tree/LeafManager.h

```diff
@@ -535,42 +535,6 @@ class LeafManager
         transform.run(this->leafRange(grainSize), threaded);
     }
 
-    template<typename ArrayT>
-    OPENVDB_DEPRECATED_MESSAGE("Use Tree::getNodes()") void getNodes(ArrayT& array)
-    {
-        using T = typename ArrayT::value_type;
-        static_assert(std::is_pointer<T>::value, "argument to getNodes() must be a pointer array");
-        using LeafT = typename std::conditional<std::is_const<
-            typename std::remove_pointer<T>::type>::value, const LeafType, LeafType>::type;
-
-        OPENVDB_NO_UNREACHABLE_CODE_WARNING_BEGIN
-        if (std::is_same<T, LeafT*>::value) {
-            array.resize(mLeafCount);
-            for (size_t i=0; i<mLeafCount; ++i) array[i] = reinterpret_cast<T>(mLeafs[i]);
-        } else {
-            mTree->getNodes(array);
-        }
-        OPENVDB_NO_UNREACHABLE_CODE_WARNING_END
-    }
-
-    template<typename ArrayT>
-    OPENVDB_DEPRECATED_MESSAGE("Use Tree::getNodes()") void getNodes(ArrayT& array) const
-    {
-        using T = typename ArrayT::value_type;
-        static_assert(std::is_pointer<T>::value, "argument to getNodes() must be a pointer array");
-        static_assert(std::is_const<typename std::remove_pointer<T>::type>::value,
-            "argument to getNodes() must be an array of const node pointers");
-
-        OPENVDB_NO_UNREACHABLE_CODE_WARNING_BEGIN
-        if (std::is_same<T, const LeafType*>::value) {
-            array.resize(mLeafCount);
-            for (size_t i=0; i<mLeafCount; ++i) array[i] = reinterpret_cast<T>(mLeafs[i]);
-        } else {
-            mTree->getNodes(array);
-        }
-        OPENVDB_NO_UNREACHABLE_CODE_WARNING_END
-    }
-
```

Comment:
what are these methods removed? Tree::getNodes() is clearly slower since they always involve tree-traversals where these methods make use the existing linear caching of nodes

Reply:
The simple answer is that they're being removed because they have been marked as deprecated for over two years now. :)

The longer answer is this - you are right that `Tree::getNodes()` is slightly slower for the specific use case of populating an array of LeafNode pointers from an existing LeafManager, but I hope you'll agree that this is a rare use case. In fact, none of the tools in the OpenVDB core library use this method nor any of the auxiliary plugins. In addition, the previous rationale for using this method has also now gone, because it used to be recommended to build a NodeManager from a LeafManager for improved performance. Now, construction of both LeafManagers and NodeManagers have been threaded such that it is faster to build them from the Tree directly. Finally, there is no `getNodes()` method on the NodeManager, so this method is inconsistent, unused and as I said marked deprecated.